homeassistant/components/starlink/config_flow.py

```diff
@@ -0,0 +1,77 @@
+"""Config flow for Starlink."""
+from __future__ import annotations
+
+from typing import Any
+
+from starlink_grpc import ChannelContext, GrpcError, get_id
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigFlow
+from homeassistant.const import (
+    CONF_NAME,
+    CONF_URL,
+    DATA_RATE_BITS_PER_SECOND,
+    DATA_RATE_GIGABITS_PER_SECOND,
+    DATA_RATE_MEGABITS_PER_SECOND,
+)
+from homeassistant.data_entry_flow import FlowResult
+
+from .const import CONFIG_LINK_SPEED_FORMAT, CONFIG_POLL_RATE, DOMAIN
+
+CONFIG_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_NAME, default="Starlink"): str,
+        vol.Required(CONF_URL, default="192.168.100.1:9200"): str,
+        vol.Required(CONFIG_POLL_RATE, default=1): vol.All(
+            vol.Coerce(int), vol.Range(min=1)
+        ),
+        vol.Required(
+            CONFIG_LINK_SPEED_FORMAT, default=DATA_RATE_MEGABITS_PER_SECOND
+        ): vol.In(
+            [
+                DATA_RATE_BITS_PER_SECOND,
+                DATA_RATE_MEGABITS_PER_SECOND,
+                DATA_RATE_GIGABITS_PER_SECOND,
+            ]
+        ),
```

Comment:
Can we detect this? Is this needed?

Reply:
The default link speed format is bit/s, which I don't find particularly useful. That said, I also didn't want to lock users out of betting the raw data, so I figured a config option would be best.
If you would still like it removed, I can just lock it to Mbit/s